services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/structure/directory/IndexDirectoryOnDisk.java

```diff
@@ -105,8 +106,10 @@ private DirectoryOnDisk makeSubDirectory(T index) {
   private void prepareMetaWritables(ImmutableStack<Object> indices, DirectoryOnDisk target) {
     this.metaWritables.forEach(metaWritableFunction -> {
       Writable<WritableOnDisk> newWritable = metaWritableFunction.apply(indices);
-      target.addWritable(newWritable);
-      newWritable.prepare(indices);
+      if (Objects.nonNull(newWritable)) {
```

Comment:
We should avoid introducing the "return null" code smell. Ideally, by not running into the necessity of having a _no-op_ code path or if that can not be achieved within reasonable effort, use something that is semantically richer (see `Optional`)